### ISSUE TITLE
Allow drafts to have no collection

### ISSUE THREAD

**Issue by tommoor** [outline/outline#2122]
Currently when creating a document you are first asked which collection it should go in. In some cases the choice is clear, but in others you don't want to make that decision immediately. Instead it would be better if creating a new document placed it in Drafts, and once you're ready to Publish a collection and location is chosen.

### Flow

- New document
- Document is created in drafts without collection
- "Publish" button is pressed
- New UI asks where to publish (collection and nesting). These options may be preset in the case of creating the document from somewhere like an existing document, but this would be a good chance to confirm.
 
### Considerations

- All permissions are built around collections and collection memberships right now, so the biggest concern would be changing these to allow for drafts without a collection and updating appropriate tests
- We will need to make `collectionId` on documents optional
- The breadcrumb for drafts and elsewhere in the UI will need updating to show collection optionally
- We will need to make `collectionId` on publish required as an API validation

**Comment by iamsaumya** [comment]
> New UI asks where to publish (collection and nesting). These options may be preset in the case of creating the document from somewhere like an existing document, but this would be a good chance to confirm.
 
Do you have any particular UI in mind?

**Comment by tommoor** [comment]
Needs a little thought, only thing I'm quite sure of right now is that it will be a fly-out when clicking the publish button

**Comment by iamsaumya** [comment]
Hey, I was going through the app, observing the flow for drafts, and imaging the new prescribed flow.

Few thoughts have come to my mind. Would you please take a look? 

> These options may be preset in the case of creating the document from somewhere like an existing document, but this would be a good chance to confirm.

When creating a doc from the collection or document menu, I think we as users are aware of what we are doing. However, I understand the need to change the options in the future but asking for confirmation when I created a draft from menus seems counterintuitive. 

In short, Shall we place this functionality in the menu? Because it will not hinder the experience of already determined publishers and users like me who want to change, we can always find our way to the menu options.

 ---- 
 
I observed that for nested docs, we don't show their parent in `DocumentListItem`. If we show, It will be beneficial in the case of drafts that have a parent.

Screenshot from linear - showing parent issue. 

![Screenshot from 2021-08-03 01-57-12](https://user-images.githubusercontent.com/40148448/127922692-44b088c8-dd10-4026-8535-496de25b6d44.png)

**Comment by tommoor** [comment]
I believe the key difference with Linear is that you can't nest items indefinitely, I think we should show the parent in the header breadcrumb for drafts but it's not something I want to introduce into all document list items.

This is a rough mock of the publish flow for a doc that has no collection (don't take any measurements or anything from this 😄 ) 

![image](https://user-images.githubusercontent.com/380914/127935736-b13a745e-f767-4a06-b441-08dd383b7e3a.png)

> asking for confirmation when I created a draft from menus seems counterintuitive.

Maybe, we could avoid the flyover for docs that already have a collection. I worry that this will be confusing to the user – why do I sometimes get this UI and othertimes not might not be immediately apparent. Perhaps we start with no confirmation and see how it feels in testing.

**Comment by iamsaumya** [comment]
Mock looks so good 👌🏻

**Comment by iamsaumya** [comment]
Few things were uncovered today, As usual looking forward to your review here.

- `maintainerApprovalRequired` column in the Collection model is not used in the codebase. Should we remove it?

- While deleting a template there is an incorrect message, which suggests that we cannot restore a deleted template.

- Delete a template, then open it. Templates and Trash both are highlighted in the sidebar.

- `CreateDocument` policy doesn't check for collection's policies. There are two instances in the app where we get the `New doc` button for the view-only collection. 
  
  - Search a random keyword and filter by view-only collection, you will get an option to create a doc.
  - Visit an empty view-only collection.

-  While choosing a collection for a doc, we get view-only collections in the menu. They are disabled but I think they shouldn't be there. 

- "maintainer" permission level, Is it even used in the app? Only found instances in models and tests.

**Comment by tommoor** [comment]
**`maintainerApprovalRequired` column and "maintainer" permission level** , partially developed feature that's still on the roadmap, in hindsight it shouldn't have been added so early but it's okay to leave for now.

The rest of these I think you should file bugs for 😄

**Comment by iamsaumya** [comment]
Does template always require a collection?

**Comment by tommoor** [comment]
Here with the hard questions. Templates are attached to collections mainly for permissions, I could see a world where there is no need for them in collections but in regards to this specific issue it might be easiest to keep them there.

**Comment by iamsaumya** [comment]
Let's say I delete a private collection, and all the docs are moved to trash. If I send the link to that doc to other users, they can view it. Though, that doc doesn't appear in the trash of that other user.

Do we want to let them view a doc that was previously in a private collection?

**Comment by tommoor** [comment]
With the collection soft-deleted it's awkward to continue to base permissions on it's memberships as there is no way to manage them.

You also don't want the docs to be visible to _everyone_ on the team as that seems like a way to expose previously private information.

I think in a perfect world we might track a `deletedBy` and only allow that user to view the trashed document 😬 – do you have thoughts?

**Comment by iamsaumya** [comment]
Sounds good to me 👍🏻 

I will see if I can stop the doc from getting viewed by everyone through any small modifications. If not, we will put a PR to tackle this with the `deletedBy` approach.

**Comment by iamsaumya** [comment]
<table>
	<tr> 
		<th> Light Mode </th>
		<th> Dark Mode </th>
	</tr>
<tr>
<td> <img src="https://user-images.githubusercontent.com/40148448/128922138-49a2745c-0a0b-4213-a765-a1b5ed0d11ac.png"/>
</td>
<td> <img src="https://user-images.githubusercontent.com/40148448/128922312-13e94551-1473-4b42-8205-d83da95c4f31.png"/>
</td> 
</tr>
</table>

Reusing the document move modal code, it's coming convenient here.

**Comment by apoorv-mishra** [comment]
Currently, a collection with :point_down: permissions

![Screenshot from 2022-09-27 13-50-16](https://user-images.githubusercontent.com/15694566/192474905-99456da1-a81e-4029-906e-92072181753d.png)

creates a draft doc with :point_down: permissions for the user

```
{
	"abilities": {
		"read": true,
		"download": true,
		"star": true,
		"unstar": true,
		"share": true,
		"update": true,
		"createChildDocument": false,
		"move": false,
		"pin": false,
		"unpin": false,
		"subscribe": false,
		"unsubscribe": false,
		"pinToHome": false,
		"delete": true,
		"permanentDelete": false,
		"restore": false,
		"archive": false,
		"unarchive": false,
		"unpublish": false
	}
}
```

What permissions should be granted to the user for a draft doc without any collection? IMO, to start with, `share` should be disabled with other permissions being kept as it is. We can keep relaxing more of these later as per the needs. What do you think @tommoor ?

**Comment by tommoor** [comment]
Yea, that sounds perfect.

We'll need to have individual sharing of private docs to enable the share permission on these which will need a new join table. Would love to get to that this year though.

**Comment by apoorv-mishra** [comment]
By the way, share button still appears in document header even when `"share": false`(though without the _Publish to internet_ toggle). Is that a bug or a feature?

**Comment by tommoor** [comment]
This is by design up until now as `share` represents public sharing – it's still useful to quickly grab a link for internal sharing through that dialog.

With this change it won't work of course until the above-mentioned work is done. The easiest for now is to put a guard for `document.collectionId` around rendering that UI, good spot.

**Comment by tommoor** [comment]
Updated designs…

## Publish modal

Button is disabled until a publish location is chosen, with a message in the footer. The pane is scrollable and needs to be virtualized.

<img width="692" alt="image" src="https://user-images.githubusercontent.com/380914/206825604-9acbc857-c429-4806-81fe-67a5039bb677.png">

## Selecting a collection / document

Click on any collection or document to choose where to publish current document. Blue indicates selected state.

<img width="701" alt="image" src="https://user-images.githubusercontent.com/380914/206825645-bea52438-d493-444a-b806-ee2c54752d2c.png">

## Filtering

Filter with the search, if the filter term matches a document then the tree above it should be shown.

<img width="713" alt="image" src="https://user-images.githubusercontent.com/380914/206825664-02a97c31-48ae-4e44-a4ee-b15803e9c498.png">
